.github/workflows/publish.yaml

```diff
@@ -0,0 +1,51 @@
+name: Publish a package to NPM registry
+
+on:
+  workflow_dispatch:
+    inputs:
+      plugin:
+        description: "Plugin to be published"
+        type: choice
+        required: true
+        options:
+          - "auth-basic"
+          - "auth-core"
+          - "auth-jwt"
+          - "broker-kafka"
+          - "ci-github-actions"
+          - "db-mongo"
+          - "db-mysql"
+          - "db-postgres"
+          - "deployment-helm-chart"
+          - "formatter-prettier"
+          - "nestjs-nats"
+          - "swagger-apibody"
+      publish:
+        description: "Publish package"
+        type: boolean
+        required: true
+        default: false
+
+jobs:
+  build-and-publish:
+    if: ${{ github.event.inputs.publish }} == true
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+
+      - name: Setup Node
+        uses: actions/setup-node@v3
+        with:
+          node-version: "18.x"
+          registry-url: "https://registry.npmjs.org"
+
+      - name: Install dependencies and build 🔧
+        run: npm ci && npm run build
+        working-directory: plugins/${{ github.event.inputs.plugin }}
+
+      - name: Publish package on NPM 📦
+        run: npm publish
+        working-directory: plugins/${{ github.event.inputs.plugin }}
+        env:
```

Comment:
It might be good to have a check against the branch and if it is not `master` we publish the package tagged as beta

more info : https://docs.npmjs.com/adding-dist-tags-to-packages#example